### DISCUSSION TITLE
[Enhancement] Integration with/into native apps

### DISCUSSION THREAD

**Discussion by ILAgent** [dotnet/maui#37] [General]
Imagine we have two native apps, android (kotlin) and iOS(swift), and we want to add some new shared functionality into them. Now it is impossible to integrate XF into native apps. It would be great to be able to do that.
Second point. Native bindings - very painful  way to adopt non-trivial native libraries. It would be great if you considered something easier.
Thank you!

**Comment by dhindrik** [comment]
It maybe not are the complete solution, but if you don't know about it it can be interesting to take a look at, https://github.com/mono/Embeddinator-4000

**Comment by ILAgent** [comment]
@dhindrik thank you, I know, but that tool looks unsupported and too difficult to use. I'd be happy to see a better way.

**Comment by jaydeeveloper** [comment]
Native binding is a real headache, the transformation of the output xml is poor documented and some libraries, specially SDK for IoT devices, definetly don't work. I hope microsoft fix it :)

**Comment by johannperez** [comment]
> It maybe not are the complete solution, but if you don't know about it it can be interesting to take a look at, https://github.com/mono/Embeddinator-4000

I've personally gone this path, not a nice experience nor results. We discovered that some libraries we developed for Xamarin apps were useful disregarding Xamarin, so we wanted to make them available to use on kotlin or swift. There's not much documentation than the issues on github, if you are lucky enough to have the same problems. Aside from the limitations, the resulting libraries were huge.

I think it could be super powerful to be able to export libraries from Xamarin to be used natively on kotlin/swift and from there on any other framework.
